app/src/main/java/ir/kindnesswall/view/main/more/MoreFragment.kt

```diff
@@ -4,104 +4,195 @@ import android.os.Bundle
+import androidx.compose.runtime.livedata.observeAsState
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.ColorFilter
+import androidx.compose.ui.platform.ComposeView
+import androidx.compose.ui.platform.ViewCompositionStrategy
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import com.chibatching.kotpref.livedata.asLiveData
+import com.google.android.material.composethemeadapter.MdcTheme
 import com.google.android.material.dialog.MaterialAlertDialogBuilder
+import com.skydoves.landscapist.glide.GlideImage
 import ir.kindnesswall.BaseFragment
+import ir.kindnesswall.BuildConfig
 import ir.kindnesswall.KindnessApplication
 import ir.kindnesswall.R
 import ir.kindnesswall.data.local.AppPref
 import ir.kindnesswall.data.local.UserInfoPref
 import ir.kindnesswall.data.model.PhoneVisibility
-import ir.kindnesswall.databinding.FragmentMoreBinding
 import ir.kindnesswall.utils.extentions.runOrStartAuth
 import ir.kindnesswall.utils.openSupportForm
 import ir.kindnesswall.view.main.MainActivity
 import ir.kindnesswall.view.main.addproduct.SubmitGiftViewModel
-import ir.kindnesswall.view.main.more.aboutus.AboutUsActivity
 import ir.kindnesswall.view.profile.UserProfileActivity
 import ir.kindnesswall.view.profile.blocklist.BlockListActivity
 import ir.kindnesswall.view.reviewgift.ReviewGiftsActivity
 import org.koin.android.viewmodel.ext.android.viewModel
 
-/**
- * Created by farshid.abazari since 2019-11-07
- *
- * Usage:
- *
- * How to call:
- *
- * Useful parameter:
- *
- */
-
-class MoreFragment() : BaseFragment() {
-    lateinit var binding: FragmentMoreBinding
-
+class MoreFragment : BaseFragment() {
     private val viewModel: SubmitGiftViewModel by viewModel()
 
-    override fun onCreateView(
-        inflater: LayoutInflater,
-        container: ViewGroup?,
-        savedInstanceState: Bundle?
-    ): View? {
-        binding = DataBindingUtil.inflate(inflater, R.layout.fragment_more, container, false)
-        MainActivity.liveData.observe(binding.root.context as LifecycleOwner, Observer {
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View =
+        ComposeView(inflater.context).apply {
+            layoutParams = LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.MATCH_PARENT)
+        }.also {
+            MainActivity.liveData.observe(viewLifecycleOwner) {
+                if (UserInfoPref.bearerToken.isNotEmpty())
+                    viewModel.refreshPhoneVisibility()
+            }
             if (UserInfoPref.bearerToken.isNotEmpty()) {
-                viewModel.refreshPhoneVisibility()
+                // TODO because of its mediator livedata (cold observer)
+                viewModel.phoneVisibilityLiveData.observe(viewLifecycleOwner) {}
```

Comment:
AFAIK, this doesn't do anything, am I right? If so, Can we remove this line?

Reply:
Thanks for your attention @H-Ghadirian 

Not now, but sure!
`phoneVisibilityLiveData` is like a cold observer and must be observed to do something on background.

I couldn't refactor all process of phone-visibility status changes at this time. So tried to avoid breaking its functionality.
So I postponed it by a TODO